lib/policies/exponentialRetryPolicy.ts

```diff
@@ -86,8 +86,8 @@ export class ExponentialRetryPolicy extends BaseRequestPolicy {
  * @param {RetryData} retryData  The retry data.
  * @return {boolean} True if the operation qualifies for a retry; false otherwise.
  */
-function shouldRetry(policy: ExponentialRetryPolicy, statusCode: number, retryData: RetryData): boolean {
-  if ((statusCode < 500 && statusCode !== 408) || statusCode === 501 || statusCode === 505) {
+function shouldRetry(policy: ExponentialRetryPolicy, statusCode: number | undefined, retryData: RetryData): boolean {
```

Comment:
A try can result in a transient failure s.t. there's no response object to even inspect. I'm pretty sure I've made this change before but somehow it didn't make it in.